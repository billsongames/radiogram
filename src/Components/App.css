/* body {
  background-image: url("../../src/assets/img/bg_radio.png");
  background-repeat: no-repeat;
  background-position: center top;
} */
.App {
  margin: auto;
}