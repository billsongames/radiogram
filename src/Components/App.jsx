import { React, useEffect, useState}  from 'react';

//############

import { getAuth, signInWithPopup, GoogleAuthProvider } from "firebase/auth";
//import {auth} from "../backend/firebase"

import { collection, get, query,  where, getCountFromServer, doc, setDoc, getDocs, documentId } from "firebase/firestore";
import {db} from "../backend/firebase"

//############

import Header from './Header/Header';
import Tuner from './Tuner/Tuner';
import Presets from './Presets/Presets';
import RadioPlayer from './RadioPlayer/RadioPlayer';
import Joint800px from './Joints/Joint800px'

//############

import radio_antenna from "../assets/img/radio_antenna.png"
import tuning_static from "../assets/audio/tuning-radio-7150.mp3"
import white_logo from "../assets/img/white.png"

//############

import './App.css';
import { api_test_data } from '../data/api_test_data';
import Joint_800px from './Joints/Joint800px';

const staticPlayer = new Audio(tuning_static)
staticPlayer.loop=(true)

//let staticIsPlaying = false




const App = () =>  {

  const [userID, setUserID] = useState("")
  const [tuned, setTuned] = useState(false)
  const [currentStation, setCurrentStation] = useState(
    {
      name: "Select a station...",
      favicon: white_logo
    })
  const [newStation, setNewStation] = useState([])
  const [presets, setPresets] = useState([])



  const handleLogin = () => {
    const provider = new GoogleAuthProvider();
    const auth = getAuth();
    signInWithPopup(auth, provider)
    .then((result) => {
      // This gives you a Google Access Token. You can use it to access the Google API.
      const credential = GoogleAuthProvider.credentialFromResult(result);
      const token = credential.accessToken;
      // The signed-in user info.
      const user = result
      setUserID(result.user.email)


      //const user = result

      // IdP data available using getAdditionalUserInfo(result)
      // ...
      }).catch((error) => {
        // Handle Errors here.
        const errorCode = error.code;
        const errorMessage = error.message;
        // The email of the user's account used.
        const email = error.customData.email;
        // The AuthCredential type that was used.
        const credential = GoogleAuthProvider.credentialFromError(error);
        // ...
        });
  }

  const handleLogout = () => {
    window.FB.logout()
    setUserID("")
  }

  const  handleStationLogoClick = event => {
    console.log(event.target.dataset.tags)
    if (event.target.id === currentStation.id){
      return
    } else {
        staticPlayer.play()
        setTuned(false)
//        staticIsPlaying = true

        setNewStation({
          id: event.target.id,
          name: event.target.name,
          favicon: event.target.src,
          urlResolved: event.target.dataset.urlresolved,
          tags: event.target.dataset.tags
        })

        setCurrentStation({
          id: event.target.id,
          name: "Tuning...",
          favicon : radio_antenna,
          urlResolved: event.target.dataset.urlresolved,
          tags: event.target.dataset.tags
        })
      }
  }

  const handleStationTuned = () => {
    staticPlayer.pause()
//    staticIsPlaying = false
    setCurrentStation(newStation)
    setTuned(true)
    }

  const handlePresetSaveClicked = async (event, stationInfo) => {
    event.preventDefault()
      const newSavedPreset = {
        id: currentStation.id,
        name: currentStation.name,
        favicon: currentStation.favicon,
        urlResolved: currentStation.urlResolved,
        tags: currentStation.tags
      }

      setPresets([...presets, newSavedPreset])
  }  

  




  useEffect(() => {
    async function checkIfUserExists() {
      if (userID) {

// CREATE THE QUERY TO COUNT MATCHING DB ENTRIES

        const coll = collection(db, "users")
        const q = query(coll, where(documentId(), "==", `${userID}`))
        
// RUN THE QUERY

        const querySnapshot = await getCountFromServer(q)

        if (querySnapshot.data().count === 0) {
          console.log("No user exists")
          await setDoc(doc(coll, `${userID}`), {
            presets: []
          })
        }
      }
    }  

    checkIfUserExists()
  }, [userID])


  useEffect(() => {
    const timer = setTimeout(() => {
      async function populatePresets() {
        if (userID){

// CREATE THE QUERY TO GET MATCHING DB ENTRIES

        const coll = collection(db, "users");
        const q = query(coll, `${userID}`)
        
// RUN THE QUERY

        const querySnapshot = await getDocs(q)
        querySnapshot.forEach((doc) => {
          setPresets(doc.data().presets);
        })
      }
    }    

    populatePresets()
  }, 500)
  return () => clearTimeout(timer)
  }, [userID])




  useEffect(() => {
    const timer = setTimeout(() => {
      async function writePresets() {
        if (userID){

          // CREATE THE QUERY TO COUNT MATCHING DB ENTRIES
          
          const coll = collection(db, "users");
          const q = query(coll, `${userID}`)
                  
          // RUN THE QUERY
          await setDoc(doc(coll, `${userID}`), {
            presets: presets
          })
        }
      }
      writePresets()
    }, 1000)
    return () => clearTimeout(timer)
    },[userID, presets])






  return (
    
    <div className='App'>
      <div className="top-section">
        <Header
          userID={userID}
          handleLogin={handleLogin}
          handleLogout={handleLogout}
          handleStationLogoClick={handleStationLogoClick}
        />
        <Tuner onStationLogoClick = {handleStationLogoClick}/>

        <Joint800px/>

        <RadioPlayer
          tuned={tuned}
          userID={userID}
          currentStation = {currentStation}
          onStationTuned = {handleStationTuned}
          onPresetSaveClicked={handlePresetSaveClicked}
          presets={presets}
        />
      </div>

      <Joint800px/>

        {userID
        ?
        <Presets
//          userID={userID}
          presets={presets}
          onStationLogoClick={handleStationLogoClick}
          onPresetSaveClicked={handlePresetSaveClicked}
          />
        :
        <></>
        }
    </div>
    
  );
}

export default App