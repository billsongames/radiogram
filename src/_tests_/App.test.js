import { render } from '@testing-library/react';
import App from "../../src/Components/App"

describe("App render", () => {

  xtest('App renders correctly', () => {
    const { asFragment } =  render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
})