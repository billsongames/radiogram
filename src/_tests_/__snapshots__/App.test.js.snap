// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="header-container"
      >
        <div
          class="header__top-row"
        >
          <div
            class="header__logo"
          >
            <img
              alt="RadioPlayer logo"
              src="logo_radiogram_2.png"
            />
          </div>
          <div
            class="header__clock"
          >
            <time
              datetime="1700584678748"
            >
              16:37
            </time>
          </div>
          <div
            class="header__login"
          >
            <div>
              <div
                class="login"
              >
                <div
                  class="login__text"
                >
                  Log in with Google
                </div>
                <svg
                  class="login__button"
                  enable-background="new 0 0 48 48"
                  fill="currentColor"
                  height="36px"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: whitesmoke;"
                  version="1.1"
                  viewBox="0 0 48 48"
                  width="36px"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <path
                    d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
                    fill="#FFC107"
                  />
                  <path
                    d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
                    fill="#FF3D00"
                  />
                  <path
                    d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
                    fill="#4CAF50"
                  />
                  <path
                    d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
                    fill="#1976D2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="joint800"
      />
      <div>
        <div
          class="eq-player-section"
        >
          <div
            class="eq_graph"
          >
            <div
              class="equaliser-container"
            >
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
            </div>
          </div>
          <div
            class="radio-player"
          >
            <div
              class="display-station-container"
            >
              <div>
                <img
                  alt="Select a station..."
                  class="display-station__logo"
                  src="white.png"
                />
              </div>
              <div
                class="display-station__info"
              >
                <div
                  class="display-station__name"
                >
                  Select a station...
                </div>
                <div
                  class="save_info"
                >
                  Log in to save presets
                </div>
              </div>
            </div>
            <div
              class="audioplayer-container"
            >
              <div
                aria-label="Audio player"
                class="rhap_container rhap_loop--off rhap_play-status--paused "
                role="group"
                tabindex="0"
              >
                <audio
                  preload="auto"
                />
                <div
                  class="rhap_main rhap_stacked-reverse"
                >
                  <div
                    class="rhap_progress-section"
                  />
                  <div
                    class="rhap_controls-section"
                  >
                    <div
                      class="rhap_main-controls"
                    >
                      <button
                        aria-label="Play"
                        class="rhap_button-clear rhap_main-controls-button rhap_play-pause-button"
                        type="button"
                      >
                        <span />
                      </button>
                    </div>
                    <div
                      class="rhap_time rhap_current-time"
                      id="rhap_current-time"
                    >
                      --:--
                    </div>
                    <div
                      class="rhap_volume-controls"
                    >
                      <div
                        class="rhap_volume-container"
                      >
                        <button
                          aria-label="Mute"
                          class="rhap_button-clear rhap_volume-button"
                          type="button"
                        >
                          <span />
                        </button>
                        <div
                          aria-label="Volume control"
                          aria-valuemax="100"
                          aria-valuemin="0"
                          aria-valuenow="100"
                          class="rhap_volume-bar-area"
                          role="progressbar"
                          tabindex="0"
                        >
                          <div
                            class="rhap_volume-bar"
                          >
                            <div
                              class="rhap_volume-indicator"
                              style="left: 100.00%; transition-duration: 0s;"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="eq_graph"
          >
            <div
              class="equaliser-container"
            >
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
              <ol
                class="equaliser-column"
              >
                <li
                  class="colour-bar"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
                <li
                  class="led-mask"
                />
              </ol>
            </div>
          </div>
        </div>
        <div
          class="joint800"
        />
        <div
          class="tuner__container"
        >
          <div
            class="tuner__title"
          >
            <div>
              STEREO TUNER
            </div>
            <div>
              <form
                class="tuner__search-form"
              >
                <input
                  class="search-form__input"
                  placeholder="Search station names..."
                  type="text"
                  value=""
                />
                <button
                  class="search-form__button"
                  type="submit"
                >
                  Go
                </button>
              </form>
            </div>
          </div>
          <div
            class="tuner__main"
          >
            <div
              class="tuner-filter__container"
            >
              <div
                class="tuner-filter__item"
                id="Pop"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    POP
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Rock"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    ROCK
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Metal"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    METAL
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Dance"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    DANCE
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Jazz"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    JAZZ
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Electro"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    ELECTRO
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Blues"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    BLUES
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Funk"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    FUNK
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Alternative"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    ALTERNATIVE
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
              <div
                class="tuner-filter__item"
                id="Classical"
              >
                <button
                  class="filter__button"
                  type="button"
                >
                  <div
                    class="filter__button-top"
                  >
                    CLASSICAL
                  </div>
                  <div
                    class="filter__button-bottom"
                  />
                  <div
                    class="filter__button-base"
                  />
                </button>
              </div>
            </div>
            <div
              class="tuner-carousel"
            >
              <div
                class="react-multi-carousel-list  "
                dir="ltr"
              >
                <ul
                  class="react-multi-carousel-track "
                  style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="joint800"
        />
        <div
          class="presets-container"
        >
          <div
            class="presets__title"
          >
            PRESETS
          </div>
          <div
            class="presets__grid"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class=""
        style="position: fixed; left: 0px; top: 0px; width: 100%; height: 100%; z-index: 999; background-color: rgba(0, 0, 0, 0.3);"
      >
        <div
          class="CookieConsent"
          style="align-items: baseline; background: rgb(43, 55, 59); color: white; display: flex; flex-wrap: wrap; justify-content: space-between; left: 0px; position: fixed; width: 100%; z-index: 999; font-size: 24px; bottom: 0px;"
        >
          <div
            class=""
            style="flex: 1 0 300px; margin: 15px;"
          >
            This website uses cookies to improve your listening experience.  
            <a
              class="cookies__privacy-policy"
              href="https://billsongames.weebly.com/privacy.html"
              target="blank"
            >
              Privacy Policy
            </a>
          </div>
          <div
            class=""
          >
            <button
              aria-label="Accept cookies"
              class=""
              id="rcc-confirm-button"
              style="background: whitesmoke; border: 0px; border-radius: 0px; box-shadow: none; color: black; cursor: pointer; flex: 0 0 auto; padding: 5px 10px; margin: 15px; font-size: 24px;"
            >
              Understood
            </button>
            <button
              aria-label="Decline cookies"
              class=""
              id="rcc-decline-button"
              style="background: red; border: 0px; border-radius: 0px; box-shadow: none; color: rgb(229, 229, 229); cursor: pointer; flex: 0 0 auto; padding: 5px 10px; margin: 15px; font-size: 24px;"
            >
              No way!
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
